### ISSUE TITLE
Section concurrency that says...

### ISSUE THREAD

**Issue by markusa** [markusa/ietf-multipath-dccp#171]
..., in some way, implementers SHOULD NOT do this unless they know what they are doing, perhaps outlining how this can be dangerous if you don't understand your traffic, etc.

https://mailarchive.ietf.org/arch/msg/tsvwg/ArSW5EsL3pEqj0zOJlB38OY7iMw/

**Comment by markusa** [comment]
### "Concurrent path usage" or better??? "Path usage concepts"

MP-DCCP can be configured to follow several concepts for path usage. A first is path mobility, which provides the use of a single path with a seamless handover function to continue the connection when the currently used path is deemed unsuitable for service delivery. Another is concurrent path utilization, which allows multiple path resources to be aggregated for higher throughput. These different path utilization concepts make MP-DCCP suitable for end-to-end implementation over the Internet or in controlled environments such as Hybrid Access or 5G ATSSS.
Using the existing Linux reference implementation [https://multipath-dccp.org] both concepts, path mobility and concurrent path usage demonstrate their functionality. However, using concurrent path usage over the Internet can lead to further implications. The choice of (coupled) congestion control, scheduler, and possible reordering function has performance and fairness consequences. Since this needs further investigation, it is strongly recommended that concurrent path usage over the Internet be used cautiously.

**Comment by veselinrak** [comment]
This is fine for me. Do we want to specify 'two' instead of 'several' path usage concepts at the start? 

Slighly modified text with some minor language correction suggestions: 

MP-DCCP can be configured to follow several concepts for path usage. The first is the path mobility concept, which provides the use of a single path with a seamless handover function to continue the connection when the currently used path is deemed unsuitable for service delivery. Another is the concurrent path utilization concept, which allows multiple path resources to be aggregated for higher throughput. These different path utilization concepts make MP-DCCP suitable for end-to-end implementation over the Internet or in controlled environments such as Hybrid Access or 5G ATSSS.

Using the existing Linux reference implementation [https://multipath-dccp.org] the functionality of both the path mobility and the concurrent path usage concepts can be demonstrated. However, using concurrent path usage over the Internet can have further implications. The choice of (coupled) congestion control, scheduler, and possible reordering function has performance and fairness consequences. Since this needs further investigation, it is strongly recommended that concurrent path usage over the Internet SHOULD be used cautiously.

**Comment by markusa** [comment]
@gorryfair 
We followed Martin's suggested step 4b from to shift from EXP to STD: https://mailarchive.ietf.org/arch/msg/tsvwg/ArSW5EsL3pEqj0zOJlB38OY7iMw/


> Stepping back from the question of how to spell this in documents, what I would like is for the non-concurrent cases to be standards track (assuming they are otherwise mature enough) while implementers are warned away from the concurrent use case unless they "know what they are doing".
> 
> **4b. Another would be a PS document with a section concurrency that says, in some way, implementers SHOULD NOT do this unless they know what they are doing, perhaps outlining how this can be dangerous if you don't understand your traffic, etc.**

**Comment by gorryfair** [comment]
Yes, this approach is OK, but if we are heading for PS the review level is high, and the separation of topics into subsections is important - at least to me.

**Comment by markusa** [comment]
Does the following structure seems suitable to you? Text I will optimize later.

### 3.9 Path usage concepts

MP-DCCP can be configured to follow several concepts for path usage. The first is the path mobility concept, which provides the use of a single path with a seamless handover function to continue the connection when the currently used path is deemed unsuitable for service delivery. Another is the concurrent path utilization concept, which allows multiple path resources to be aggregated for higher throughput. These different path utilization concepts make MP-DCCP suitable for end-to-end implementation over the Internet or in controlled environments such as Hybrid Access or 5G ATSSS.

Using the existing Linux reference implementation [https://multipath-dccp.org] the functionality of both the path mobility and the concurrent path usage concepts can be demonstrated.

#### 3.9.1 Path mobility
bla bla


#### 3.9.2 Concurent path usage

Using concurrent path usage over the Internet can have implications. The choice of (coupled) congestion control, scheduler, and possible reordering function has performance and fairness consequences. Since this needs further investigation, it is strongly recommended that concurrent path usage over the Internet SHOULD NOT be used unless

**Comment by gorryfair** [comment]
Good text can be hard, sorry.

I wonder if it could be more:

3.9 Path usage concepts

MP-DCCP can be configured to realise one of several concepts for path usage, via XXXX. 

Selecting an appropriate method can allow MP-DCCP to realise different path utilization concepts that make MP-DCCP suitable for end-to-end implementation over the Internet or in controlled environments such as Hybrid Access or 5G ATSSS.

3.9.1 Path mobility

This specification specifies the path mobility concept, which provides the use of a single path with a seamless handover function to continue the connection when the currently used path is deemed unsuitable for service delivery. bla bla -
<<<have a look at failover text for SCTP???>>>

This is supported in the current Linux reference implementation [https://multipath-dccp.org]. 

3.9.2 Concurrent path usage

The method could be used to support concurrent path utilization concept, which allows multiple path resources to be aggregated for higher throughput.

Using concurrent path usage over the Internet can have implications. The choice of (coupled) congestion control, scheduler, and possible reordering function has performance and fairness consequences. Since this needs further investigation, it is strongly recommended that concurrent path usage over the Internet SHOULD NOT be used unless....

 <<I think it is better to say this is not specified in this specific document for use across the general Internet. Such an IETF specification will need to consider ....

This is also supported in the current Linux reference implementation [https://multipath-dccp.org].

**Comment by markusa** [comment]
@gorryfair after further alignment between the authors, this is our latest version. Do you have further comments or do you think it is mature enough to put the topic on the mailinglist?

### 3.9 Path usage strategies
 
MP-DCCP can be configured to realise one of several strategies for path usage, via selecting one DCCP subflow of the multiple DCCP subflows within a MP-DCCP connection for data transmission. This can be a dynamic process further facilitated by the means of DCCP and MP-DCCP defined options such as path preference using MP-PRIO, adding or removing DCCP subflows using MP_REMOVEADDR, MP_ADDADDR or DCCP-Close/DCCP-Reset and also path metrics such as packet-loss-rate, CWND or RTT provided by the Congestion Control Algorithm.
 
Selecting an appropriate method can allow MP-DCCP to realise different path utilization strategies that make MP-DCCP suitable for end-to-end implementation over the Internet or in controlled environments such as Hybrid Access or 5G ATSSS.
 
#### 3.9.1 Path mobility
 
The path mobility strategy provides the use of a single path with a seamless handover function to continue the connection when the currently used path is deemed unsuitable for service delivery.
 
Some of the DCCP subflows of a MP-DCCP connection might become inactive due to either the occurrence of certain error conditions (e.g., DCCP timeout, packet loss threshold, RTT threshold, closed/removed) or adjustments from the MP-DCCP user.
 
When there is outbound data to send and the primary path becomes inactive (e.g., due to failures) or de-prioritized, the MP-DCCP endpoint SHOULD try to send the data through an alternate path with a different source or destination address (depending on the point of failure), if one exists. This process SHOULD respect the path prio configured by MP_PRIO or if not available pick the most divergent source-destination pair from the original used source-destination pair.
 
Note: Rules for picking the most divergent source-destination pair are an implementation decision and are not specified within this document.
 
Path mobility is supported in the current Linux reference implementation [https://multipath-dccp.org/].
 
#### 3.9.2 Concurrent path usage
 
This method could be used to support a concurrent path utilization strategy, which allows multiple path resources to be aggregated for higher throughput.
 
Compared to the path mobility strategy, the selection of DCCP flows is a per-packet decision and part of the multipath scheduling process which is out of scope of this specification.
 
Concurrent path usage over the Internet can have implications. The choice of (coupled) congestion control, scheduler, and possible reordering function has performance and fairness consequences. Since this needs further investigation, it is strongly recommended that concurrent path usage over the Internet SHOULD NOT be used unless the implementer knows about the implications at and between the MP-DCCP termination points.
 
Concurrent path usage is also supported in the current Linux reference implementation [https://multipath-dccp.org/].
